### PR TITLE
Test to checks Snapshot events are emitted

### PR DESCRIPTION
**Pull request by ninokeldishvili**
## Motivation
Convert snapshot hardhat tests into foundry 

## Solution
write new tests in foundry

## Checks

By submitting this for review, I'm confirming I've done the following:
- [x] made this PR as small as possible
- [x] unit-tested any new functionality
- [x] linked any relevant issues or PRs

### CODE REVIEW

File: test/foundry/Snapshot.t.sol [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        // Prank as Alice for the transaction
+        vm.startPrank(alice);
+        OffchainAssetReceiptVault vault = OffchainAssetVaultCreator.createVault(factory, alice, assetName, assetSymbol);
+        vault.grantRole(vault.ERC20SNAPSHOTTER(), alice);
+
+        vm.expectEmit(true, true, true, true);
+        emit Snapshot(1);
+
+        vm.expectEmit(true, true, true, true);
+        emit SnapshotWithData(alice, 1, data);
```
Comment (thedavidmeister) [inline test/foundry/Snapshot.t.sol:29]: why 1?

**Review by thedavidmeister — CHANGES_REQUESTED** [review]

File: test/foundry/Snapshot.t.sol [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    function testSnapshot(uint256 fuzzedKeyAlice, bytes memory data, string memory assetName, string memory assetSymbol)
+        external
+    {
+        fuzzedKeyAlice = bound(fuzzedKeyAlice, 1, SECP256K1_ORDER - 1);
+        address alice = vm.addr(fuzzedKeyAlice);
+
+        // Prank as Alice for the transaction
+        vm.startPrank(alice);
+        OffchainAssetReceiptVault vault = OffchainAssetVaultCreator.createVault(factory, alice, assetName, assetSymbol);
+        vault.grantRole(vault.ERC20SNAPSHOTTER(), alice);
```
Comment (thedavidmeister) [inline test/foundry/Snapshot.t.sol:23]: do we have a test that you can't do this without this role?

**Review by thedavidmeister — CHANGES_REQUESTED** [review]

File: test/foundry/Snapshot.t.sol (hunk 2 above)
Comment (ninokeldishvili) [reply test/foundry/Snapshot.t.sol:23]: yes, in Roles.t.sol rest

File: test/foundry/Snapshot.t.sol (hunk 1 above)
Comment (ninokeldishvili) [reply test/foundry/Snapshot.t.sol:30]: is not the snapshot ids increase, with the first value being 1?

**Review by thedavidmeister — APPROVED** [review]

**Pull request merged**